### PR TITLE
Use internal libxml errors instead of @suppression

### PR DESCRIPTION
**Pull request by lsl**
## Description
Instead of using `@` to suppress errors during `DOMDocument::loadHTML`
use `libxml_use_internal_errors` to capture and suppress the errors
instead.

If there is a custom `set_error_handler` set for your installation it
will continue to receive warnings and html validation errors from this
`loadHTML` call. If the goal is to suppress these errors we should try
to capture and drop these without polluting customized error logs.

## How has this been tested?
- Added a unit test to capture errors that would be missed by `@` suppression
- Removed the change and confirmed the test fails
- Watched the php error log of a test site using the latest posts block with the block setting of "Show post date" toggled on.

## Types of changes
Bug fix, there shouldn't be any breaking changes although some people may see less errors for broken block html or html5 features passing through this code.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

**Comment by lsl** [comment]
@ockham may want to take a look at this too - I'm not totally sure what the original suppression was for but I'm hoping this is capturing all those same errors.

File: phpunit/class-block-supported-styles-test.php
Comment (ockham) [inline phpunit/class-block-supported-styles-test.php:808]:
```suggestion
		// HTML5 elements like <time> are not supported by the DOMDocument parser used by the block supports feature.
```

**Review by ockham — APPROVED** [review]
Looking good, thanks for the fix! [One typo](https://github.com/WordPress/gutenberg/pull/24677#discussion_r480078754), otherwise good to merge (once GH Actions go all green).

**Comment by lsl** [comment]
Reminder: I don't have merge rights here.

**Comment by ockham** [comment]
Merged. Thanks again @lsl!

@jorgefilipecosta @youknowriad @mcsf Can we maybe include this in 8.9? These warning can otherwise clutter warning logs (or the frontend, if WP_DEBUG is true).

**Comment by youknowriad** [comment]
Seems like all warnings are not suppressed yet https://wordpress.org/support/topic/warning-a-non-numeric-value-encountered-31/#post-13357203 

I wonder if we should try to avoid DOMDocument and just use Regex to include the classes and styles like explored here https://github.com/WordPress/gutenberg/pull/21420/files#diff-4339a9e86722fd029c231f3fb65f0ad2R394-R412

**Pull request merged**
